### PR TITLE
CyberArk PAS - new integration

### PR DESCRIPTION
**Pull request by ohaim1008**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/17698
fixes: https://github.com/demisto/etc/issues/14367

## Description
A new integration for CyberArk PAS (privileged account security)

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS_test.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+from CyberArkPAS import Client, add_user_command, get_users_command, \
+    update_user_command, add_safe_command, update_safe_command, get_list_safes_command, get_safe_by_name_command, \
+    add_safe_member_command, update_safe_member_command, list_safe_members_command, add_account_command, \
+    update_account_command, get_list_accounts_command, get_list_account_activity_command, fetch_incidents
+from test_data.context import ADD_USER_CONTEXT, GET_USERS_CONTEXT, \
+    UPDATE_USER_CONTEXT, UPDATE_SAFE_CONTEXT, GET_LIST_SAFES_CONTEXT, GET_SAFE_BY_NAME_CONTEXT, ADD_SAFE_CONTEXT, \
+    ADD_SAFE_MEMBER_CONTEXT, UPDATE_SAFE_MEMBER_CONTEXT, LIST_SAFE_MEMBER_CONTEXT, ADD_ACCOUNT_CONTEXT, \
+    UPDATE_ACCOUNT_CONTEXT, GET_LIST_ACCOUNT_CONTEXT, GET_LIST_ACCOUNT_ACTIVITIES_CONTEXT, INCIDENTS, INCIDENTS2, \
+    INCIDENTS3, INCIDENTS4
```
Comment (bakatzir) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS_test.py:11]: let's have better names than INCIDENTS<number>

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 2]
```diff
@@ -0,0 +1,1375 @@
...
+import urllib3
+import traceback
+
+# Disable insecure warnings
+
+urllib3.disable_warnings()
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+
+def parse_date_range_expire_date(date_range):
```
Comment (bakatzir) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:16]: why not use the one we have in CSP?

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 3]
```diff
@@ -0,0 +1,1375 @@
...
+    if 0 <= score <= 35:
+        return 1
+    elif 35 < score <= 75:
+        return 2
+    elif 75 < score <= 100:
+        return 3
+    return 0
+
+
+def filter_by_score(events_data: list, score: int):
```
Comment (bakatzir) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:72]:
```suggestion
def filter_by_score(events_data: list, score: int) -> list:
```

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 4]
```diff
@@ -0,0 +1,1375 @@
...
+    start, _ = parse_date_range("7 days")
+    start_time_timestamp = str(date_to_timestamp(start))
+    security_events = client.get_security_events(start_time_timestamp)
+    # if there were security events in the last week
+    if security_events:
+        event = security_events[0]
+        if event.get("id") and event.get("type") and event.get("score"):
+            return "ok"
+        else:
+            raise Exception("Security events are not in the right format")
```
Comment (bakatzir) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:570]: more elegant to have only one return statement. WDYT?
```suggestion
        if not event.get("id") or not event.get("type") or no event.get("score"):
            raise Exception("Security events from CyberArk PAS are missing mandatory fields.")
```

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 5]
```diff
@@ -0,0 +1,1375 @@
...
+    # if there were no security events in the last week but the http request seceded
```
Comment (bakatzir) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:571]: can rm

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 6]
```diff
@@ -0,0 +1,1375 @@
...
+        'cyberark-pas-credentials-change-in-vault-only': change_credentials_in_vault_only_command,
+        'cyberark-pas-credentials-verify': verify_credentials_command,
+        'cyberark-pas-credentials-reconcile': reconcile_credentials_command,
+
+        'cyberark-pas-security-events-get': get_security_events_command,
+    }
+
+    try:
+        client = Client(server_url=url, username=username, password=password, use_ssl=use_ssl, proxy=proxy,
+                        max_fetch=int(max_fetch))
```
Comment (bakatzir) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:1348]: let's have a defualt rounding down of 200 as Anar suggested.

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py (hunk 4 above)
Comment (ohaim1008) [reply Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:570]: looks much better

**Comment by ShirleyDenkberg** [comment]
@ohaim1008 Review is done.

**Pull request merged**